rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -298,11 +342,70 @@ rmw_fastrtps_shared_cpp::destroy_participant(CustomParticipantInfo * participant
     RMW_SET_ERROR_MSG("participant_info is null");
     return RMW_RET_ERROR;
   }
-  Domain::removeParticipant(participant_info->participant);
-  delete participant_info->listener;
-  participant_info->listener = nullptr;
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(participant_info->participant_, RMW_RET_ERROR);
```

Comment:
@MiguelCompany same comment about not checking for states that imply preconditions were not respected.